cdap-api/src/main/java/co/cask/cdap/internal/dataset/DatasetRuntimeContext.java

```diff
@@ -78,17 +78,14 @@ public static Cancellable setContext(DatasetRuntimeContext context) {
     final Thread callerThread = Thread.currentThread();
     final DatasetRuntimeContext oldContext = CONTEXT_THREAD_LOCAL.get();
     CONTEXT_THREAD_LOCAL.set(context);
-    return new Cancellable() {
-      @Override
-      public void cancel() {
-        // This must be called from the same thread that call setContext
-        Thread currentThread = Thread.currentThread();
-        if (currentThread != callerThread) {
-          LOG.warn("Cancel is called from different thread. Expected {}, actual: {}", callerThread, currentThread);
-          return;
-        }
-        CONTEXT_THREAD_LOCAL.set(oldContext);
+    return () -> {
```

Comment:
does this only work for interfaces w/ 1 method?
Otherwise, what identifies which method this implements?

Reply:
Inferred from return type of the method.

Reply:
and yes, only for interface with one method.